lib/google_maps_geocoder/google_maps_geocoder.rb

```diff
@@ -52,8 +54,7 @@ class GoogleMapsGeocoder
   #   chez_barack = GoogleMapsGeocoder.new '1600 Pennsylvania Ave'
   def initialize(data)
     @json = data.is_a?(String) ? json_from_url(data) : data
-    fail "Geocoding \"#{data}\" exceeded query limit! Google returned...\n"\
-         "#{@json.inspect}" if @json.blank? || @json['status'] != 'OK'
+    handle_exception(@json)
```

Comment:
Please rename the method and only call it if you need to:

```ruby
handle_error if @json.blank? || @json['status'] != 'OK'
```

 Note you *don't* have to pass in the JSON as a method argument because `@json` is an instance variable.